app/uk/gov/hmrc/pillar2/handlers/Pillar2ErrorHandler.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.pillar2.handlers
+
+import play.api.http.HttpErrorHandler
+import play.api.libs.json.Json
+import play.api.mvc.Results._
+import play.api.mvc.{RequestHeader, Result}
+import uk.gov.hmrc.pillar2.models.errors._
+
+import javax.inject.Singleton
+import scala.concurrent.Future
+
+@Singleton
+class Pillar2ErrorHandler extends HttpErrorHandler {
+
+  override def onClientError(request: RequestHeader, statusCode: Int, message: String): Future[Result] =
+    Future.successful(BadRequest(Json.toJson(Pillar2ApiError(statusCode.toString, message))))
```

Comment:
this doesn't have to be BadRequest. For example, if the third party calls an incorrect endpoint, it would cause a 404, but be transformed here into a 400

Reply:
Interesting...

turns out, a 404 is returned:
```
{
  "code": "404",
  "message": ""
}
```

This must be only if our controller returns a client error.

Reply:
On this, as discussed we went with just providing badRequest but unfortunately that kills a test on another suite that dies on a 404 which has me worrying about us messing with the output of other endpoints if i'm being honest.

the test is on RegistrationCacheControllerSpec:
```
"return 404 when the data doesn't exist" in new Setup {
        when(mockRegistrationCacheRepository.getLastUpdated(eqTo("foo"))(any[ExecutionContext]())) thenReturn Future.successful {
          None
        }

        val request: FakeRequest[AnyContentAsEmpty.type] = FakeRequest(POST, routes.RegistrationCacheController.lastUpdated("id").url)
        val result:  Future[Result]                      = route(application, request).value

        status(result) mustBe NOT_FOUND
      }
```

Reply:
looks like we'll have to simply return the status code with the message. Nothing else we can do here